### PR TITLE
[IGA] some cleanup

### PR DESCRIPTION
**Pull request by philbucher**
this PR 
- removes some unneeded includes / cleans them up
- adds missing overrides
- removes compiler warnings abt comparison signed-int <=> unsigned-int

### CODE REVIEW

File: applications/IgaApplication/iga_application.h [hunk 1]
```diff
@@ -76,7 +76,7 @@ class KratosIgaApplication
     ///@name Operations
     ///@{
 
-    virtual void Register();
+    void Register() override;
```
Comment (oberbichler) [inline applications/IgaApplication/iga_application.h:78]: [Also in the Application-Template?](https://github.com/KratosMultiphysics/Kratos/blob/a741949882c360ae3f738839a793b6c1a449f5a7/kratos/templates/application#L98)

Comment (philbucher) [reply applications/IgaApplication/iga_application.h:78]: Oh I thought I had changed that already 
Will do in a separate PR

**Review by tteschemacher — APPROVED** [review]
classical unsigned int <-> int discussion. Thanks for the fixes.

File: applications/IgaApplication/iga_application.h (hunk 1 above)
Comment (oberbichler) [reply applications/IgaApplication/iga_application.h:78]: 👍 Maybe also the Info-stuff at the end? (actually I am just not sure... if it is ok then I can also do the job)

**Pull request merged**